pkgs/development/tools/build-managers/meson/gir-fallback-path.patch

```diff
@@ -1,17 +1,24 @@
+diff --git a/mesonbuild/modules/gnome.py b/mesonbuild/modules/gnome.py
+index 1c6952df7..9efbdcee0 100644
 --- a/mesonbuild/modules/gnome.py
 +++ b/mesonbuild/modules/gnome.py
-@@ -913,6 +913,14 @@ class GnomeModule(ExtensionModule):
+@@ -922,6 +922,19 @@ class GnomeModule(ExtensionModule):
+             raise MesonException('fatal_warnings keyword argument must be a boolean')
          if fatal_warnings:
              scan_command.append('--warn-error')
- 
-+        if len(set([girtarget.get_custom_install_dir() for girtarget in girtargets])) > 1:
```

Comment:
Upstream added type hinting to the get_custom_install_dir function so now it is a optional type containing a tupple (str,bool). This change also changed the default/null case; previously the default return value would be [None] but now the default case is simple an empty list [].

If a python-guru knows how to handle a empty set within a list comprehension, we could certainly make this less verbose. Also if anyone has any preference on how the patch is written such as how the empty list check is done or anything like that, I'm all ears.

Reply:
We could use `[girtarget.get_custom_install_dir()[0] if girtarget.get_custom_install_dir() is not None for girtarget in girtargets]`

Reply:
We do not even need the square brackets as Python supports set comprehensions.

Reply:
Thanks I forgot a clause could be added to the end of the list/set comprehension. I went with just `set(girtarget.get_custom_install_dir()[0] for girtarget in girtargets if girtarget.get_custom_install_dir())`.